src/main/java/seedu/address/logic/parser/property/FindCommandParser.java

```diff
@@ -1,31 +1,73 @@
      */
     public FindCommand parse(String args) throws ParseException {
-        String trimmedArgs = args.trim();
-        if (trimmedArgs.isEmpty()) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        ArgumentMultimap argMultimapBoth = ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_ADDRESS);
+        ArgumentMultimap argMultimapBothTwo = ArgumentTokenizer.tokenize(args, PREFIX_ADDRESS, PREFIX_NAME);
+        ArgumentMultimap argMultimapName = ArgumentTokenizer.tokenize(args, PREFIX_NAME);
+        ArgumentMultimap argMultimapAddress = ArgumentTokenizer.tokenize(args, PREFIX_ADDRESS);
+        LandlordName extractedName;
+        Address extractedAddress;
+
+        if (arePrefixesPresent(argMultimapBoth, PREFIX_NAME, PREFIX_ADDRESS)
+                || arePrefixesPresent(argMultimapBothTwo, PREFIX_ADDRESS, PREFIX_NAME)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        }
+
+        if (arePrefixesPresent(argMultimapName, PREFIX_NAME) && argMultimapName.getPreamble().isEmpty()) {
+            extractedName = ParserUtil.parseLandlordName(argMultimapName.getValue(PREFIX_NAME).get());
+            String trimmedExtractedName = extractedName.toString().trim();
+            if (trimmedExtractedName.isEmpty()) {
+                throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+            }
+            String[] keywords = trimmedExtractedName.split("\\s+");
+            return new FindCommand(new LandlordNameContainsKeywordsPredicate(Arrays.asList(keywords)));
+        }
+
+        if (arePrefixesPresent(argMultimapAddress, PREFIX_ADDRESS) && argMultimapAddress.getPreamble().isEmpty()) {
+            extractedAddress = ParserUtil.parseAddress(argMultimapAddress.getValue(PREFIX_ADDRESS).get());
+            String trimmedExtractedAddress = extractedAddress.toString().trim();
+            if (trimmedExtractedAddress.isEmpty()) {
+                throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+            }
+            String[] keywords = trimmedExtractedAddress.split("\\s+");
+            return new FindCommand(new AddressContainsKeywordsPredicate(Arrays.asList(keywords)));
         }
+        throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+    }
 
-        String[] addressKeywords = trimmedArgs.split("\\s+");
-        return new FindCommand(new AddressContainsKeywordsPredicate(Arrays.asList(addressKeywords)));
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
```

Comment:
got ensure to check for either /s or /a is present?

Reply:
this function is used to check for presence of n/ or a/ for finding property command